components/sftp/sftp.app.js

```diff
@@ -0,0 +1,53 @@
+const ssh2SftpClient = require('ssh2-sftp-client');
+
+module.exports = {
+  type: 'app',
+  app: 'sftp',
+  props: {
+    db: '$.service.db'
```

Comment:
I realized after looking at this again that this won't work. We currently require props be declared only in the component (you can define share `propDefinitions` within the app file, just not instances of `props`). 

We're talking about this more internally - I totally see where you're coming with this approach, and I think it does make sense for the app file to be able to manage its own local state via `db`. It's just not currently possible.